RealmSwift-swift2.0/Error.swift

```diff
@@ -0,0 +1,108 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2014 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import Realm
+
+
+/**
+Enumeration that describes the error codes within the Realm error domain.
+The values can be used to catch a variety of _recoverable_ errors, especially those
+happening when initializing a Realm instance.
+
+    let realm: Realm?
+    do {
+        realm = Realm()
+    } catch RealmSwift.Error.IncompatibleLockFile() {
+        print("Realm Browser app may be attached to Realm on device?")
+    }
+
+*/
+public enum Error: ErrorType {
+    /// :nodoc:
+    public var _code: Int {
+        return rlmError.rawValue
+    }
+
+    /**
+    - returns: the RLMError value, which can be used to derive the error's code.
+    */
+    internal var rlmError: RLMError {
+        switch self {
+        case .Fail:
+            return RLMError.Fail
+        case .FileAccessError:
+            return RLMError.FileAccessError
+        case .FilePermissionDenied:
+            return RLMError.FilePermissionDenied
+        case .FileExists:
+            return RLMError.FileExists
+        case .FileNotFound:
+            return RLMError.FileNotFound
+        case .IncompatibleLockFile:
+            return RLMError.IncompatibleLockFile
+        }
+    }
+
+    /// :nodoc:
+    public var _domain: String {
+        return RLMErrorDomain
+    }
+
+    /**
+    - returns: error thrown by RLMRealm if no other specific error is returned when a realm is opened.
```

Comment:
Since this is Realm Swift, please change 'RLMRealm' references to 'Realm' instead.